projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java

```diff
@@ -1,24 +1,590 @@
+import com.projects.opentrain.gtfs_pipeline.CalendarProtos.Calendar;
+import com.projects.opentrain.gtfs_pipeline.FareAttributesProtos.FareAttributes;
+import com.projects.opentrain.gtfs_pipeline.FareRulesProtos.FareRules;
+import com.projects.opentrain.gtfs_pipeline.RoutesProtos.Route;
+import com.projects.opentrain.gtfs_pipeline.ShapesProtos.Shape;
+import com.projects.opentrain.gtfs_pipeline.StopTimeListProtos.StopTime;
+import com.projects.opentrain.gtfs_pipeline.StopsProtos.Stop;
+import com.projects.opentrain.gtfs_pipeline.TranslationsProtos.Translation;
+import com.projects.opentrain.gtfs_pipeline.TripsProtos.Trip;
+import com.google.protobuf.TextFormat;
+import java.io.Reader;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.omg.CORBA.portable.InputStream;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.nio.file.StandardOpenOption;
 import java.util.Arrays;
 import java.util.List;
+
+import javax.management.RuntimeErrorException;
+
 import java.lang.String;
+import java.lang.Iterable;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import java.util.Iterator;
+import java.lang.Object;
+import java.io.OutputStream;
+import java.io.FileOutputStream;
+import java.io.FileNotFoundException;
+import java.io.Reader;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileInputStream;
 
 class DataHandler {
-      public static void main(String[] args) throws IOException {
-        String output = "";
-        Path outputPath = Paths.get(args[args.length-1]);
-        for (String path : args) {
-            // TODO: Read the files in a streaming manner since they are large
-            Path curPath = Paths.get(path);
-            output += curPath.getFileName() + " lines: ";
-            Files.write(outputPath, new String(curPath.getFileName()+"\n").getBytes(), StandardOpenOption.APPEND, StandardOpenOption.CREATE);
-        }
-        System.out.print(output);
+
+  public static void saveAgency(Path csvPath, String path) {
+     
+      // where we serialize the messages
+     try (FileOutputStream output = new FileOutputStream(path);)
+
+      {
+        CSVParser parser = CSVParser.parse(csvPath, StandardCharsets.UTF_8, CSVFormat.DEFAULT
+        .withHeader(
+          "agency_id",
+          "agency_name",
+          "agency_url",
+          "agency_timezone",
+          "agency_lang",
+          "agency_phone",
+          "agency_fare_url"));
+      // Creat iterator - to create from each record a message
+      Iterator<CSVRecord> iterator = parser.iterator();

```

Comment:
I think a cleaner way to do this would be:
```
for (CSVRecord record : parser) {
  ...
}
```
You would still have to skip the first one, so maybe do this:
```
boolean isHeader = true;
for (CSVRecord record : parser) {
  if (isHeader) {
    isHeader = false;
    continue;
  }
  ...
}
```